server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -465,6 +465,10 @@ public Decision canRemain(IndexMetadata indexMetadata, ShardRouting shardRouting
         );
     }
 
+    private static boolean ignoreDiskWatermarksForShard(RoutingAllocation allocation, ShardRouting shardRouting) {
+        return shardRouting.role().isPromotableToPrimary() || allocation.metadata().index(shardRouting.index()).ignoreDiskWatermarks();
```

Comment:
`shardRouting.role().isPromotableToPrimary()` should return true for `Role.DEFAULT` as well. I do not think this is intended. Should it be `shardRouting.role().isSearchable() == false` instead?

Reply:
Oh, stupid me, I should have checked 🤦🏻

Reply:
I pushed b550560b951201043a92d82a8605ce9338b52137 which I think is correct